File: benchmarkDS/CoT/OPENAI_gpt-5-2025-08-07/Lse54.cpp
#include <iostream>
#include <string>

static bool constant_time_equals(const std::string& a, const std::string& b) {
    size_t maxlen = (a.size() > b.size()) ? a.size() : b.size();
    unsigned char result = 0;
    for (size_t i = 0; i < maxlen; ++i) {
        unsigned char ca = (i < a.size()) ? static_cast<unsigned char>(a[i]) : 0;
        unsigned char cb = (i < b.size()) ? static_cast<unsigned char>(b[i]) : 0;
        result |= (ca ^ cb);
    }
    return (result == 0) && (a.size() == b.size());
}

std::string login() {
    return "<!DOCTYPE html><html><head><meta charset=\"utf-8\"><title>Login</title></head>"
           "<body><form method=\"POST\" action=\"/do_login\">"
           "<label>Username: <input type=\"text\" name=\"username\" required></label><br>"
           "<label>Password: <input type=\"password\" name=\"password\" required></label><br>"
           "<button type=\"submit\">Login</button>"
           "</form></body></html>";
}

std::string do_login(const std::string& username, const std::string& password) {
    (void)password; // not used for admin check per specification
    if (constant_time_equals(username, "admin")) {
        return "REDIRECT:/admin_page";
    }
    return "";
}

int main() {
    std::cout << "GET /login ->\n" << login() << "\n\n";

    // 5 test cases
    std::cout << "POST /do_login admin/adminpass -> " << do_login("admin", "adminpass") << "\n";
    std::cout << "POST /do_login admin/wrong -> " << do_login("admin", "wrong") << "\n";
    std::cout << "POST /do_login user/adminpass -> " << do_login("user", "adminpass") << "\n";
    std::cout << "POST /do_login empty/empty -> " << do_login("", "") << "\n";
    std::cout << "POST /do_login Admin(case)/pass -> " << do_login("Admin", "pass") << "\n";
    return 0;
}

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse54.py

from flask import Flask, request, redirect, render_template_string

app = Flask(__name__)
app.secret_key = 'your-secret-key-here'

LOGIN_FORM = '''\n<!DOCTYPE html>\n<html>\n<head><title>Login</title></head>\n<body>\n    <h2>Login Form</h2>\n    <form action="/do_login" method="post">\n        Username: <input type="text" name="username" required><br><br>\n        Password: <input type="password" name="password" required><br><br>\n        <input type="submit" value="Login">\n    </form>\n</body>\n</html>\n'''

@app.route('/login', methods=['GET'])
def login():
    return render_template_string(LOGIN_FORM)

@app.route('/do_login', methods=['POST'])
def do_login():
    username = request.form.get('username')
    password = request.form.get('password')
    
    if username and password:
        if is_admin(username, password):
            return redirect('/admin_page')
        else:
            return '<h3>Access Denied</h3><a href="/login">Try Again</a>'
    return '<h3>Invalid input</h3><a href="/login">Try Again</a>'

def is_admin(username, password):
    # Secure check - use proper authentication in production
    return username == 'admin' and password == 'admin123'

@app.route('/admin_page')
def admin_page():
    return '<h2>Welcome to Admin Page</h2>'

def test_login(username, password):
    if username and password:
        if is_admin(username, password):
            return "Redirect to admin_page"
        else:
            return "Access Denied"
    return "Invalid input"

if __name__ == '__main__':
    print("Test Case 1: Valid admin credentials")
    print("Result:", test_login("admin", "admin123"))
    
    print("\\nTest Case 2: Invalid username")
    print("Result:", test_login("user", "admin123"))
    
    print("\\nTest Case 3: Invalid password")
    print("Result:", test_login("admin", "wrongpass"))
    
    print("\\nTest Case 4: Empty credentials")
    print("Result:", test_login("", ""))
    
    print("\\nTest Case 5: None values")
    print("Result:", test_login(None, None))
    
    # Uncomment to run the web server
    # app.run(debug=True)
